components/templates/McGonagall/McGonagall.tsx

```diff
@@ -194,10 +350,16 @@ class McGonagall extends React.Component {
     newState.actions.forEach((action) => {
       // If the action is executable, execute it
       action.exec &&
```

Comment:
I'd prefer if this checked the type of exec rather than only testing its existence. This gives a little more confidence (at runtime) that `exec` can be invoked. Though based on the type signature of `ActionObject` this is probably not necessary. This stems from my dislike of truthy checks in TypeScript but you're welcome to ignore this one!

```suggestion
      typeof action.exec === 'function' &&
```

Reply:
I'm probably going to leave this one as is for now but I'd love to know more about your "dislike of truthy checks in TS".

Reply:
It mostly boils down to my feeling that the advantages of TypeScript come from requiring the developer to be explicit about their expectations regarding what is contained in a value and truthy checks ignore that. In this example you're relying on the implied type of the `action.exec` property to not change. If the type of `action.exec` does change this as written will fail at runtime and it won't necessarily be clear to the new developer about why. Encoding the expectation explicitly, if the type of `action.exec` were to be changed away from a function I believe `tsc` would at least warn and say "this can never be true".